### PR TITLE
update docs links

### PR DESCRIPTION
**Pull request by tzemanovic**
This PR updates links to the Book and Rustdoc to https://proptest-rs.github.io/proptest and https://docs.rs/proptest, respectively.

Additionally, there used to be a separate build for no-std deployed at https://altsysrq.github.io/rustdoc/proptest-nostd/latest/proptest/. Instead of 2 docs builds, the commit 274f6ad0e6da816dbf7cee9d5782df0770ebeb4e exposes features to the regular docs.rs build using the guide from https://users.rust-lang.org/t/how-to-document-optional-features-in-api-docs/64577/3

edit: docs.rs issue has been fixed

~~Do not merge yet - There seems to be an issue with the latest release on https://docs.rs/proptest/latest/proptest/ - for some reason the macros are not showing so the links from the book to the macros rustdoc are broken. They are showing in the prev version and when I build docs locally so not sure what's up with that~~

### CODE REVIEW

File: proptest/Cargo.toml [hunk 1]
```diff
@@ -121,3 +121,7 @@ optional = true
 
 [dev-dependencies]
 regex = "1.0"
+
+[package.metadata.docs.rs]
+all-features = true
+rustdoc-args = ["--cfg", "docsrs"]
```
Comment (rex-remind101) [inline proptest/Cargo.toml:127]: missing newline it looks like

**Review by rex-remind101 — COMMENTED** [review]
looks great overall, but not sure what you mean by "for some reason the macros are not showing...", do you have a link?

**Comment by rex-remind101** [comment]
i see, wiled guess but is the addition of `cfg_attr` in various places causing that disappearance? if not then i agree that in issue should be filed with docs.rs

**Review by matthew-russo — APPROVED** [review]

**Review by cameron1024 — APPROVED** [review]
Thanks :grin:

**Comment by rex-remind101** [comment]
update with `master` and ci should get unblocked

**Pull request merged**